website/content/en/docs/tasks/deprov-nodes.md

```diff
@@ -1,36 +1,84 @@
+Whether through node expiry or manual deletion, Karpenter seeks to follow graceful termination procedures as described in Kubernetes [Graceful node shutdown](https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdow) documentation.
 
-Generally, pod workloads may be configured with `.spec.minAvailable` and/or `.spec.maxUnavailable`. Karpenter provisions nodes to accommodate these constraints. 
+{{% alert title="Note" color="primary" %}}
+By adding the finalizer, Karpenter improves the default Kubernetes process of node deletion.
+When you run `kubectl delete node` on a node without a finalizer, the node gets deleted in the API server but the instance keeps running.
+The kubelet isn’t watching for its own existence, so if a node is deleted the kubelet doesn’t terminate itself.
+All the pod objects get deleted a minute later, from a garbage collection process because the pods’ nodes are gone.
+{{% /alert %}}
 
-## Emptiness
+## What can cause deprovisioning to fail?
 
-Karpenter will delete nodes (and the instance) that are considered empty of pods. Daemonset pods are not included in this calculation. 
+There are a few cases where requesting to deprovision a Karpenter node will fail. These include Pod Disruption Budgets and pods that have the `do-not-evict` annotation set.
 
-## Expiry
+### Disruption budgets
 
-Nodes may be configured to expire. That is, a maximum lifetime in seconds starting with the node joining the cluster. Review the `ttlSecondsUntilExpired` field of the [provisioner API](../../provisioner/).
+Karpenter respects Pod Disruption Budgets (PDBs) when provisioning nodes and can prevent a node from being deprovisioned to maintain those disruption budgets.
+Kubernetes PDBs let you specify how much of a Deployment, ReplicationController, ReplicaSet, or StatefulSet must be protected from disruptions when pod eviction requests are made. 
+
+PDBs can be used to strike a balance by protecting the application's availability while still allowing a cluster operator to manage the cluster.
+Here is an example where the pods matching the label `myapp` will block node termination if evicting the pod would reduce the number of available pods below 4.
+
+```bash
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: myapp-pdb
+spec:
+  minAvailable: 4
+  selector:
+    matchLabels:
+      app: myapp
+```
+
+You can set `minAvailable` or `maxUnavailable` as integers or as a percentage.
+Review what [disruptions are](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/), and [how to configure them](https://kubernetes.io/docs/tasks/run-application/configure-pdb/).
+
+### Pod set to do-not-evict
+
+Karpenter will not drain a node if a pod exists with the annotation `karpenter.sh/do-not-evict`.
+The node will be cordoned to prevent additional work from scheduling.
```

Comment:
It will also not drain any pods until the `do-not-evict` pod is removed.
https://github.com/aws/karpenter/blob/main/pkg/controllers/termination/terminate.go#L68-L81 shows that if there is a pod that has a `do-not-evict` annotation, the draining mechanism will not evict any pods.

Additionally, it might be worth noting that if a `do-not-evict` pod somehow gets scheduled to a node mid-draining, the remainder of pods will still evict, but that pod will block termination until it is removed. This is a small edge case, so it might not be worth noting.